Feat: Consumers key concept/reference page (#7512)
Kong/docs.konghq.com: app/_src/gateway/key-concepts/consumers.md

---
title: Consumers
concept_type: reference
---

## What is a consumer?

A consumer typically refers to an entity that consumes or uses the APIs managed by {{site.base_gateway}}. 
Consumers can be applications, services, or users who interact with your APIs. 

> Since they are not always human, {{site.base_gateway}} calls them consumers, as they "consume" the service.

Reviewer: ```suggestion
Since they are not always human, {{site.base_gateway}} calls them consumers, because they "consume" the service.
```

> {{site.base_gateway}} allows you to define and manage consumers, apply access control policies, and monitor their usage of APIs.

Reviewer: ```suggestion
{{site.base_gateway}} allows you to define and manage consumers, apply access control policies, and monitor their API usage.
```

Consumers are essential for controlling access to your APIs, tracking usage, and ensuring security.
They are identified by key authentication, OAuth, or other authentication and authorization mechanisms. 

> For example, adding a Basic Auth plugin to a service or route allows it to identify a consumer, or block access if credentials are bad.

Reviewer: At first I thought linking to the plugin might be a good idea, but I think it might detract from what you are trying to teach.

Author: yep, the specific auth plugin is not important in this case, it could be any auth plugin. linking off to it would be an unnecessary distraction.

> For example, adding a Basic Auth plugin to a service or route allows it to identify a consumer, or block access if credentials are bad.

Reviewer: ```suggestion
For example, adding a Basic Auth plugin to a service or route allows it to identify a consumer, or block access if credentials are invalid.
```

> You can either rely on {{site.base_gateway}} as the primary datastore, or you can map the consumer list 

Reviewer: ```suggestion
You can choose to use {{site.base_gateway}} as the primary datastore for consumers, or you can map the consumer list 
```

> to your database to keep consistency between {{site.base_gateway}} and your existing primary datastore.

Reviewer: ```suggestion
to an existing database to keep consistency between {{site.base_gateway}} and your existing primary datastore.
```

> By attaching a plugin directly to a consumer, you can control things on a consumer level - for example, rate limits.

Reviewer: ```suggestion
By attaching a plugin directly to a consumer, you can manage specific controls at the consumer level, such as rate limits.
```

{% mermaid %}
flowchart LR

A(Consumer entity)
B(Auth plugin)
C[Upstream service]

Client --> A
subgraph id1[{{site.base_gateway}}]
direction LR
A--Credentials-->B
end

B-->C
{% endmermaid %}

## Use cases for consumers

> The following is a short list of common use cases for consumers:

Reviewer: ```suggestion
The following are examples of common use cases for consumers:
```
